### PR TITLE
[Synapse] Add pipeline, linked service, trigger, notebook management cmdlets

### PR DESCRIPTION
**Pull request by wonner**
## Description

This change add pipeline, linked service, trigger, notebook management cmdlets. Here is the commandlet review PR:https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/694

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
Please resolve the conflicts and make sure your PR can build. Thanks

File: src/Synapse/Synapse/help/Set-AzSynapseNotebook.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+Required: True
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Executors
+Number of executors to be allocated in the specified Spark pool for the job.
```
Comment (isra-fel) [inline src/Synapse/Synapse/help/Set-AzSynapseNotebook.md:120]: I forgot to mention that parameter name should use singular form, see https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#singularity

Suggest `-ExecutorCount`

File: src/Synapse/Synapse/Models/SynapseAnalyticsArtifactsClient.cs [hunk 2]
```diff
@@ -88,9 +89,11 @@ public void DeletePipeline(string pipelineName)
             _pipelineClient.StartDeletePipeline(pipelineName);
         }
 
-        public CreateRunResponse CreatePipelineRun(string pipelineName, string referencePipelineRunId, bool? isRecovery, string startActivityName, IDictionary<string, object> parameters)
+        public string CreatePipelineRun(string pipelineName, string referencePipelineRunId, bool? isRecovery, string startActivityName, IDictionary<string, object> parameters)
```
Comment (isra-fel) [inline src/Synapse/Synapse/Models/SynapseAnalyticsArtifactsClient.cs:92]: Changing the return type to string would affect `InvokeAzureSynapsePipeline.cs`, which was designed to return a `PSCreateRunResponse`

Comment (wonner) [reply src/Synapse/Synapse/Models/SynapseAnalyticsArtifactsClient.cs:92]: It's a bug need to fix. In the next released package of Artifacts, the return type will be CreateRunResponse.

**Comment by msJinLei** [comment]
@wonner Please resolve the conflicts.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**